scipy/optimize/_constraints.py

```diff
@@ -316,8 +299,15 @@ class BoxConstraint:
         stay feasible along the iterations and each does not.
         False by default.
     """
-    def __init__(self, kind, enforce_feasibility=False):
-        self.kind = kind
+    def __init__(self, lb=-np.inf, ub=np.inf, enforce_feasibility=False):
+        lb = np.asarray(lb)
+        ub = np.asarray(ub)
+        if np.all(lb == -np.inf):
```

Comment:
Hi Nikolay,

I had time to take a look on your implementation. Really clean and nice.

Just about the future, it may worth considering saving ``ub`` and ``lb``. This way it will be easier to access them, for instance for using on finite differences.

This way, it may worth considering replacing this first few lines with:
```
self.lb = np.asarray(lb)
self.ub = np.asarray(ub)
```